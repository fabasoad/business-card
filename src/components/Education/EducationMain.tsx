import * as React from 'react'
import { useTranslation } from 'react-i18next'

import EducationItem from './EducationItem'

export default function EducationMain() {
  const { t } = useTranslation()
  return (
    <div className="light-component" id="education">
      <div className="container">
        <div className="section-title text-center">
          <h2>{t('business-card-education-title')}</h2>
          <hr />
        </div>
        <div className="row">
          <div className="col-lg-12">
            <ul className="timeline">
              <li>
                <EducationItem from={2009} to={2010} title={t('business-card-education-university-master')} />
              </li>
              <li className="timeline-inverted">
                <EducationItem from={2005} to={2009} title={t('business-card-education-university-bachelor')} />
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  )
}
