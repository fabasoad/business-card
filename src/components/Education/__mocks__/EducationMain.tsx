import * as React from 'react'

export default function EducationMain() {
  return (
    <div data-testid="EducationMain"></div>
  )
}
