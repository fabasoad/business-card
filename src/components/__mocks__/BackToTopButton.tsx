import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
