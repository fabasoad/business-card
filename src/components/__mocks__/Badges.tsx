import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
