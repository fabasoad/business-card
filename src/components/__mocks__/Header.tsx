import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
