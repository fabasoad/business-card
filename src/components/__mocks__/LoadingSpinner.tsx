import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
