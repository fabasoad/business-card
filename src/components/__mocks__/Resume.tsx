import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
