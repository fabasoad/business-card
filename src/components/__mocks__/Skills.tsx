import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
