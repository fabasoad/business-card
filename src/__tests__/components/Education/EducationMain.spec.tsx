/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import EducationMain from '../../../components/Education/EducationMain'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render EducationItem correctly', () => {
  const wrapper: ShallowWrapper = shallow(<EducationMain />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(3)
  expect(tMock).toHaveBeenCalledWith('business-card-education-title')
  expect(tMock).toHaveBeenCalledWith('business-card-education-university-master')
  expect(tMock).toHaveBeenCalledWith('business-card-education-university-bachelor')
})
