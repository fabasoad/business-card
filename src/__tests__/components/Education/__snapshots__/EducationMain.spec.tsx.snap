// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EducationItem correctly 1`] = `
<div
  className="light-component"
  id="education"
>
  <div
    className="container"
  >
    <div
      className="section-title text-center"
    >
      <h2>
        business-card-education-title
      </h2>
      <hr />
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-12"
      >
        <ul
          className="timeline"
        >
          <li>
            <EducationItem
              from={2009}
              title="business-card-education-university-master"
              to={2010}
            />
          </li>
          <li
            className="timeline-inverted"
          >
            <EducationItem
              from={2005}
              title="business-card-education-university-bachelor"
              to={2009}
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
